compiler/rustc_session/src/session.rs

```diff
@@ -978,7 +978,7 @@ impl Session {
     pub fn needs_plt(&self) -> bool {
         // Check if the current target usually needs PLT to be enabled.
         // The user can use the command line flag to override it.
-        let needs_plt = self.target.needs_plt;
+        let needs_plt = self.target.arch != "x86_64";
```

Comment:
This should still be based on a target property. Just inverted from `needs_plt` to `no_plt_by_default` or something.

Reply:
I did that, but I realized as I did it that this might invalidate existing target json files. Is that something I need to worry about?

(I'm also only about 70% sure I did this right, so please give it careful scrutiny)